code/workspaces/auth-service/src/dataaccess/userRolesRepository.js

```diff
@@ -9,14 +9,37 @@ function UserRoles() {
 // In line with other repository functions, this returns a promise,
 // but the model document is first converted to an object so the spread operator works as expected.
 function getRoles(userId) {
-  return UserRoles().findOne({ userId }).exec().then(document => document.toObject());
+  return UserRoles().findOne({ userId }).exec()
+    .then((document) => {
+      try {
+        return document.toObject();
+      } catch (err) {
+        return addEmptyRecordForNewUser(userId);
+      }
+    });
 }
 
 function getProjectUsers(projectKey) {
   const query = { 'projectRoles.projectKey': { $eq: projectKey } };
   return UserRoles().find(query).exec();
 }
 
+async function addEmptyRecordForNewUser(userId) {
+  // setOnInsert used to prevent accidental override of entry
+  const query = { userId };
+  const user = {
+    userId,
+    projectRoles: [],
+    instanceAdmin: false,
+  };
+  await UserRoles().findOneAndUpdate(
+    query,
+    { $setOnInsert: user },
+    { upsert: true, setDefaultsOnInsert: true, runValidators: true },
+  );
```

Comment:
If we leave this as `findOneAndUpdate` I would expect this to need a `.exec()` chained to the `findOneAndUpdate` and for it to return the promise rather than the user object.

Having said that, I don't know if you intentionally returned fast without waiting for the promise to resolve as in this case that might be OK...though might be confusing later.